content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,200 @@
+        length=32,
+        iterations=100000,
+        salt=salt
+    )
+
+    key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+    crypt = Fernet(key)
+    
+    # Decrypts the wallet's private key
+    _seed = crypt.decrypt(_seed)
+    
+    # Initialize XRPL wallet using decrypted private key
+    _wallet = wallet.Wallet(seed=_seed.decode(), sequence=0)
+
+    validated_seq = _ledger_seq
+    _wallet.sequence = _wallet_seq
+    
+    # Construct Payment transaction
+    my_tx_payment = Payment(
+        account=_wallet.classic_address,
+        amount=xrp_to_drops(xrp=_xrp_amount),
+        destination=_destination,
+        last_ledger_sequence=validated_seq + 100, # +100 to catch up with the ledger when we transmit the signed tx blob to Machine 2
+        sequence=_wallet.sequence,
+        fee="10"
+    )
+
+    # Signs transaction and displays the signed_tx blob in QR code
+    # Scan the QR code and transmit the signed_tx blob to an online machine (Machine 2) to relay it to the XRPL
+    my_tx_payment_signed = safe_sign_transaction(transaction=my_tx_payment, wallet=_wallet)
+
+    img = qrcode.make(my_tx_payment_signed.to_dict())
+    img.save(get_path("/Wallet/transactionID.png"))
+    image = Image.open(get_path("/Wallet/transactionID.png"))
+    image.show()
+
+
+def get_path(file):
+    """
+    Get path (filesystem management)
+    """
+
+    global File_
+    # Checks what OS is being us
+    OS = platform.system()
+    usr = Path.home()
+
+    # Get PATH format based on the OS
+    if OS == "Windows":
+        File_ = PureWindowsPath(str(usr) + file)
+    if OS == "Linux":
+        File_ = PurePath(str(usr) + file)
+
+    return str(File_)
+
+
+def main():
+    global File, Path_
+    
+    # Gets the machine's operating system (OS)
+    OS = platform.system()
+    usr = Path.home()
+    if OS == "Windows":
+        # If it's Windows, use this path:
+        File = PureWindowsPath(str(usr) + '/Wallet')
+        Path_ = str(PureWindowsPath(str(usr)))
+    if OS == "Linux":
+        # If it's Linux, use this path:
+        File = PurePath(str(usr) + '/Wallet')
+        Path_ = str(PurePath(str(usr)))
+        
+    # If the Wallet's folder already exists, continue on
+    if os.path.exists(File):
+        while True:
+            ask = int(input("\n 1. Transact XRP"
+                            "\n 2. Generate an XRP wallet (read only)"
+                            "\n 3. Showcase XRP Wallet Address (QR Code)"
+                            "\n\n Enter Index: "
+                            ))
+
+            if ask == 1:
+                password = str(input("             Enter Password: "))
+                amount = float(input("\n           Enter XRP To Send: "))
+                destination = input("\n        Enter Destination: ")
+                wallet_sequence = int(input("\n    Enter Wallet Sequence: "))
+                ledger_sequence = int(input("\n    Enter Ledger Sequence: "))
```

Comment:
Getting the Ledger Sequence is non-trivial and should have a hint about how to get it I think. (My wording is probably not the best, but this is an example of what that could look like)
```suggestion
                ledger_sequence = int(input("Look up the latest ledger index on livenet.xrpl.org and put it here\n    Enter Ledger Sequence : "))
```

Reply:
Ye I think that's a good addition, since this'll be used in a dev environment: I'll refer the testnet.